src/model/channel/message.rs

```diff
@@ -796,16 +796,18 @@ impl Message {
         #[cfg(feature = "cache")]
         {
             if let Some(cache) = cache_http.cache() {
-                utils::user_has_perms_cache(
-                    cache,
-                    self.channel_id,
-                    self.guild_id,
-                    Permissions::MANAGE_MESSAGES,
-                )?;
-
-                if self.author.id != cache.current_user_id() {
-                    return Err(Error::Model(ModelError::NotAuthor));
-                }
+                if let Err(Error::Model(ModelError::InvalidPermissions(..))) =
+                    utils::user_has_perms_cache(
+                        cache,
+                        self.channel_id,
+                        self.guild_id,
+                        Permissions::MANAGE_MESSAGES,
+                    )
+                {
+                    if self.author.id != cache.current_user_id() {
+                        return Err(Error::Model(ModelError::NotAuthor));
+                    }
+                };
```

Comment:
I think the conditionals should be swapped in order if you're nesting them like this. Since, we don't care about permissions if we're the author. Also, if `user_has_perms_cache` returns an error that isn't `InvalidPermissions`, we should propagate it up as normal.